src/components/Row.jsx

```diff
@@ -33,6 +39,20 @@ export default class Row extends Component {
     ));
   }
 
+  shouldDisplayControls(columnControls) {
+    let showControls = false;
+    columnControls.forEach((controls) => {
+      controls.controls.forEach((control) => {
```

Comment:
Can we refactor this logic to use less number of nested foreach? may be by moving the logic of setting showControls  from inside the map function of getControlsByColumn?

Reply:
In getControlsByColumn method, we need to prepare the complete data and then depends on all columns data, we can know whether to hide/show some element.